node/buildutils.js

```diff
@@ -57,59 +61,86 @@ exports.getExternals = function () {
     }
 }
 
-var downloadFile = function (url) {
-    // validate parameters
-    if (!url) {
-        throw new Error('Parameter "url" must be set.');
-    }
+exports.getExternalsAsync = getExternalsAsync
 
-    // short-circuit if already downloaded
-    var scrubbedUrl = url.replace(/[/\:?]/g, '_');
-    var targetPath = path.join(downloadPath, 'file', scrubbedUrl);
-    var marker = targetPath + '.completed';
-    if (!test('-f', marker)) {
-        console.log('Downloading file: ' + url);
 
+
+/**
+ * @deprecated This method uses library which is not prefered to use on production
+ */
+
+exports.getExternals = function () {
+    var result = false;
+    getExternalsAsync().then(t => result = true);
+    deasync.loopWhile(function () { return !result });
+    return result;
+}
+
+
+var downloadFileAsync = async function (url, fileName) {
+    return new Promise(async (resolve, reject) => {
```

Comment:
probably we can use just async/await here

Reply:
Fixed